drift-client/src/main/java/io/airlift/drift/client/ThriftMethodMetadata.java

```diff
@@ -0,0 +1,255 @@
+                    parameterName = thriftField.name();
+                }
+            }
+            if (parameterId == Short.MIN_VALUE) {
+                parameterId = (short) (index + 1);
+            }
+            if (parameterName == null) {
+                parameterName = parameterNames.get(index);
+            }
+
+            Type parameterType = parameterTypes[index];
+
+            ThriftType thriftType = catalog.getThriftType(parameterType);
+
+            ThriftInjection parameterInjection = new ThriftParameterInjection(parameterId, parameterName, index, parameterType);
+
+            if (parameterRequiredness == Requiredness.UNSPECIFIED) {
+                // There is only one field injection used to build metadata for method parameters, and if a
+                // single injection point has UNSPECIFIED requiredness, that resolves to NONE.
+                parameterRequiredness = Requiredness.NONE;
+            }
+
+            ThriftFieldMetadata fieldMetadata = new ThriftFieldMetadata(
+                    parameterId,
+                    isLegacyId,
+                    false,
+                    parameterRequiredness,
+                    parameterIdlAnnotations,
+                    new DefaultThriftTypeReference(thriftType),
+                    parameterName,
+                    THRIFT_FIELD,
+                    ImmutableList.of(parameterInjection),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty()
+            );
+            builder.add(fieldMetadata);
+        }
+        parameters = builder.build();
+
+        exceptions = buildExceptionMap(catalog, thriftMethod);
+
+        this.oneway = thriftMethod.oneway();
+    }
+
+    public String getName()
+    {
+        return name;
+    }
+
+    public ThriftType getReturnType()
+    {
+        return returnType;
+    }
+
+    public List<ThriftFieldMetadata> getParameters()
+    {
+        return parameters;
+    }
+
+    public Map<Short, ThriftType> getExceptions()
+    {
+        return exceptions;
+    }
+
+    public Method getMethod()
+    {
+        return method;
+    }
+
+    public boolean getOneway()
+    {
+        return oneway;
+    }
+
+    private ImmutableMap<Short, ThriftType> buildExceptionMap(ThriftCatalog catalog, ThriftMethod thriftMethod)
+    {
+        ImmutableMap.Builder<Short, ThriftType> exceptions = ImmutableMap.builder();
+        Set<Type> exceptionTypes = new HashSet<>();
+        int customExceptionCount = 0;
+
+        if (thriftMethod.exception().length > 0) {
+            for (ThriftException thriftException : thriftMethod.exception()) {
+                exceptions.put(thriftException.id(), catalog.getThriftType(thriftException.type()));
+                checkState(exceptionTypes.add(thriftException.type()), "ThriftMethod.exception contains more than one value for %s", thriftException.type());
```

Comment:
Make this `checkArgument`. Change message to "ThriftMethod exception list contains ..."